### DISCUSSION TITLE
general question

### DISCUSSION THREAD

**Discussion by ahmad681** [idaholab/moose#22084] [Q&A General] [answered]
Hi, I have a question on how to utilize the MOOSE software. 
I am running kinetic monte carlo simulation of precipitate in alloys. I want to include the elastic lattice mismatch of coherent interfaces between phases. So I want to run kMC, get output and analyze if needed to have elastic strain energy solution at lattice points. if yes, then pass on the output in a continuum media fashion to MOOSE and solve the mechanical equilibrium problem, evaluate the stresses and strains, and the elastic strain energy, then feed back to the KMC input to be used for the probability calculations. Is there any helpful source on how to connect external code for specific calculations to MOOSE? Many thanks in advance.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Q&A general is likely the wrong category for this. Is this a phase field simulation issue? Which domain does this belong in?
@dschwen do we have kinetic monte carlo

Guillaume

**Reply by ahmad681** [reply]
oops sorry about that, no, it's actually just kinetic monte carlo. so every atom in the space has a direction it can diffuse to. I select an atom and direction to move to based on kMC algorithm. So I have the code working for that. Now if I want to include elastic stresses, it would affect the hopping rate calculations. So I was wondering if I can couple MOOSE with my kMC code. thanks a lot

**Reply by GiudGiud** [reply]
is it an external code? or is it built in MOOSE?
Are you going to transfer a field of value over a mesh?

**Reply by ahmad681** [reply]
It is an external code I wrote. I guess the hypothetical idea is to transfer field output from kMC to the mesh in MOOSE in every kMC step.

**Reply by GiudGiud** [reply]
ok you will want to look at coupling in external application. The external problem is the official way to do this but there are other ways
https://mooseframework.inl.gov/application_development/moose_wrapped_apps.html

Please have a look at these posts:
https://github.com/idaholab/moose/discussions/16438 (external petsc problem example)
https://github.com/idaholab/moose/discussions/19413 (working with an external library)
https://github.com/idaholab/moose/discussions/16497 (examples of external apps at the bottom)

**Reply by ahmad681** [reply]
Awesome, I will check those out. Many thanks
